fix from upstream

http://bugs.gentoo.org/153782

===================================================================
RCS file: /sources/cpio/cpio/src/copyout.c,v
retrieving revision 1.19
retrieving revision 1.20
diff -u -r1.19 -r1.20
--- cpio/src/copyout.c	2006/09/27 09:28:50	1.19
+++ cpio/src/copyout.c	2006/10/24 10:48:04	1.20
@@ -806,6 +806,7 @@
 		    free (link_name);
 		    continue;
 		  }
+		link_name[link_size] = 0;
 		cpio_safer_name_suffix (link_name, false,
 					!no_abs_paths_flag, true);
 		link_size = strlen (link_name);
